### DISCUSSION TITLE
patch-transformer supporting multivariate predict univariate

### DISCUSSION THREAD

**Discussion by Essie0715** [Nixtla/neuralforecast#599] [Q&A] [answered]
Hi Everyone, 

I have got a quick question. Does the current patch-transformer support multivariate predict univariate in the current version?

**Comment by kdgutier** [comment]
Hey @Essie0715, 

There are two things you can do for a multivariate PatchTST:
1. You can adapt PatchTST to have multivariate outputs (not inputs) with the [Mixture Mesh probabilities](https://nixtla.github.io/neuralforecast/losses.pytorch.html#gaussian-mixture-mesh-gmm).
2. You can feed other series temporal data through the exogenous features parameters `futr_exog_list, hist_exog_list`.

We have one dedicated multivariate model with the [StemGNN](https://nixtla.github.io/neuralforecast/models.stemgnn.html).

**Reply by Essie0715** [reply]
Hi @kdgutier

Thank you for your reply. My task is to use multiple time series (historical temperature, wind, humidity, etc) as input features and a Single output forecasting target (predict the future temperature). The original PatchTST is multiple input and multiple output models. But I want to make it a single output model only.

**Reply by kdgutier** [reply, marked as answer]
In our current version of PatchTST we treat it as a univariate model (single series output).
Exogenous features support for the PatchTST and other Transformers is in our list of priorities.

Here is a tutorial for exogenous variables: https://nixtla.github.io/neuralforecast/examples/exogenous_variables.html
